src/corecel/data/LdgIterator.hh

```diff
@@ -0,0 +1,218 @@
+    {
+        return LDGLoadPolicy::read(ptr_ + n);
+    }
+    //!@}
+
+    // Conversion operators
+    //!@{
+    //! \name Conversion operators
+    CELER_CONSTEXPR_FUNCTION explicit operator pointer() const noexcept
+    {
+        return ptr_;
+    }
+    CELER_CONSTEXPR_FUNCTION explicit operator bool() const noexcept
+    {
+        return ptr_ != nullptr;
+    }
+    //!@}
+
+  private:
+    pointer ptr_{nullptr};
+};
+
+//---------------------------------------------------------------------------//
+// FREE FUNCTIONS
+//---------------------------------------------------------------------------//
+
+//!@{
+//! RandomAccessIterator requirements
+template<class T>
+CELER_CONSTEXPR_FUNCTION bool
+operator==(LdgIterator<T> const& lhs, LdgIterator<T> const& rhs) noexcept
+{
+    using pointer = typename LdgIterator<T>::pointer;
+    return static_cast<pointer>(lhs) == static_cast<pointer>(rhs);
+}
+template<class T>
+CELER_CONSTEXPR_FUNCTION bool
+operator!=(LdgIterator<T> const& lhs, LdgIterator<T> const& rhs) noexcept
+{
+    return !(lhs == rhs);
+}
+template<class T>
+CELER_CONSTEXPR_FUNCTION bool
+operator<(LdgIterator<T> const& lhs, LdgIterator<T> const& rhs) noexcept
+{
+    using pointer = typename LdgIterator<T>::pointer;
+    return static_cast<pointer>(lhs) < static_cast<pointer>(rhs);
+}
+template<class T>
+CELER_CONSTEXPR_FUNCTION bool
+operator>(LdgIterator<T> const& lhs, LdgIterator<T> const& rhs) noexcept
+{
+    return rhs < lhs;
+}
+template<class T>
+CELER_CONSTEXPR_FUNCTION bool
+operator<=(LdgIterator<T> const& lhs, LdgIterator<T> const& rhs) noexcept
+{
+    return !(lhs > rhs);
+}
+template<class T>
+CELER_CONSTEXPR_FUNCTION bool
+operator>=(LdgIterator<T> const& lhs, LdgIterator<T> const& rhs) noexcept
+{
+    return !(lhs < rhs);
+}
+template<class T>
+CELER_CONSTEXPR_FUNCTION LdgIterator<T>
+operator+(LdgIterator<T> const& it,
+          const typename LdgIterator<T>::difference_type n) noexcept
+{
+    LdgIterator tmp{it};
+    return tmp += n;
+}
+template<class T>
+CELER_CONSTEXPR_FUNCTION LdgIterator<T>
+operator+(const typename LdgIterator<T>::difference_type n,
+          LdgIterator<T> const& it) noexcept
+{
+    return it + n;
+}
+template<class T>
+CELER_CONSTEXPR_FUNCTION LdgIterator<T>
+operator-(LdgIterator<T> const& it,
+          const typename LdgIterator<T>::difference_type n) noexcept
+{
+    LdgIterator<T> tmp{it};
+    return tmp -= n;
+}
+template<class T>
+CELER_CONSTEXPR_FUNCTION auto
+operator-(LdgIterator<T> const& lhs, LdgIterator<T> const& rhs) noexcept ->
+    typename LdgIterator<T>::difference_type
+{
+    using pointer = typename LdgIterator<T>::pointer;
+    return static_cast<pointer>(lhs) - static_cast<pointer>(rhs);
+}
+template<class T>
+CELER_CONSTEXPR_FUNCTION void
+swap(LdgIterator<T>& lhs, LdgIterator<T>& rhs) noexcept
+{
+    return lhs.swap(rhs);
+}
+//!@}
+
+//!@{
+//! Helper
+template<class T>
+inline LdgIterator<T> make_LdgIterator(T const* ptr) noexcept
```

Comment:
```suggestion
inline LdgIterator<T> make_ldg_iterator(T const* ptr) noexcept
```

Also the `const` here makes me think that we *should* use CTAD to construct `LdgIterator<int>` from `int const*`, and maybe a static assert to require `!is_const_v<T>`?